langserver/fs_test.go

```diff
@@ -0,0 +1,70 @@
+package langserver
+
+import (
+	"testing"
+
+	"github.com/sourcegraph/go-langserver/pkg/lsp"
+)
+
+type applyContentChangesTestCase struct {
+	code     string
+	changes  []lsp.TextDocumentContentChangeEvent
+	expected string
+}
+
+var applyContentChangesTestCases = map[string]applyContentChangesTestCase{
```

Comment:
Ah, after you extract the content change calculation, these tests look much better than my compromised one in 325afcc18412b620414884bc305f1748c70205ac. You may revert that if you please, because they basically aim to do the same thing.

Reply:
I think it's a good idea to keep it in because it asserts the behaviour on a higher level (integration test) in addition to my unit tests.